http/src/main/scala/org/http4s/blaze/http/http2/Http2FrameSerializer.scala

```diff
@@ -0,0 +1,293 @@
+package org.http4s.blaze.http.http2
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+
+import org.http4s.blaze.http.http2.Http2Settings.Setting
+import org.http4s.blaze.util.BufferTools
+
+private object Http2FrameSerializer {
+  import bits._
+
+  // Override the scala provided `require(condition, => msg)` to avoid the thunks
```

Comment:
Is this more important than getting the values (e.g., line 29)?  Could we have our cake and eat it too with a *gasp* macro?